### ISSUE TITLE
Interleaving bug in sapH

### ISSUE THREAD

**Issue by zzlei** [magnumripper/JohnTheRipper#1351]
core dump when using an interleaving factor of x8 for SHA512

**Comment by magnumripper** [comment]
```C
/*
 * Assumption is made that SIMD_COEF_32*SHA1_SSE_PARA is >= than
 * SHA256_COEF*PARA and SHA512_COEF*PARA, and that these other 2
 * will evenly divide the SIMD_COEF_32*SHA1_SSRE_PARA value.
 * Works with current code. BUT if SHA1_SSE_PARA was 3 and
 * SIMD_PARA_SHA256 was 2, then we would have problems.
 */
```
:cry:

**Comment by magnumripper** [comment]
Fixed in bd32b4b1
